tinygrad/codegen/linearizer.py

```diff
@@ -418,22 +420,22 @@ def linearize(self):
     self.applied_opts_cache = self.applied_opts[:]
     return self
 
-  def ast_parse(self, x:LazyOp, acc: List[UOp], offs:Optional[List[int]], loaded_buffers:Dict[Union[MemBuffer, ConstBuffer, LocalBuffer], List[UOp]], do_reduce=False, cache=None) -> List[UOp]: # noqa: E501
+  def ast_parse(self, x:LazyOp, accs: Dict[LazyOp, List[UOp]], offs:Optional[List[int]], loaded_buffers:Dict[Union[MemBuffer, ConstBuffer, LocalBuffer], List[UOp]], do_reduce:Optional[List[UOp]]=None, cache=None) -> List[UOp]: # noqa: E501
```

Comment:
can you rename do_reduce to match its new dtype, some ideas: reduce_accs, curr_accs, loop_accs